src/base64Utils.js

```diff
@@ -8,6 +8,9 @@
  * Decodes a base64 string.
  * @param {string} input The string to decode.
  */
+#if (MIN_IE_VERSION == TARGET_NOT_SUPPORTED || MIN_IE_VERSION >= 10) && ENVIRONMENT == 'web' // IE 10+ have atob, and so do practically all other browsers. Older node (at least 14) does not, though, nor does d8, and support in workers requires somewhat newer versions of browsers, so gate this on env=web. https://developer.mozilla.org/en-US/docs/Web/API/atob
```

Comment:
Can you put the command on the line before and split it into multiple line?

Reply:
I think in other discussion it came up that while it is awkward to put it all on a single line, the benefit is that the comment does not show up without the ifdef. That is, if we have this:
```pp
// FOO requires us to do stuff
#if FOO
..
#endif
```
then if FOO is not enabled, the output is this:
```
// FOO requires us to do stuff
```
and that comment by itself is odd. Putting the comment inside the ifdef is another option we could consider, but currently most places in the code leave it on one line AFAIK.

Reply:
(If you meant to split the logic in the if into multiple lines, that doesn't work as the simple preprocessor works line-by-line.)

Reply:
I vote for putting the commend inside the `#if` and formatting across many lines.

Reply:
I was not aware of the issue of putting comments before `#if` but I agree with you that putting the comments before can lead to confusing cases in the output.

Perhaps we should include the `#if` and `#endif` in the final output but in comments if we want to make the output JS readable with comments?   Obviously that is a separate discussion to this PR.

Reply:
Hmm, the comment inside the if still has some issues. One option:
```c
#if (MIN_IE_VERSION == TARGET_NOT_SUPPORTED || MIN_IE_VERSION >= 10) && ENVIRONMENT == 'web'
// IE 10+ have atob, and so do practically all other browsers. Older node (at
// least 14) does not, though, nor does d8, and support in workers requires
// somewhat newer versions of browsers. So if we don't support IE and we are on
// the web, we don't need a polyfill here, but otherwise we do.
// See https://developer.mozilla.org/en-US/docs/Web/API/atob
var decodeBase64 = atob;
#else
var decodeBase64 = typeof atob === 'function' ? atob : function (input) {
  .. polyfill code
```
A downside is that when we don't include the polyfill we have a long explanation about polyfilling. Or, another option:
```c
#if (MIN_IE_VERSION == TARGET_NOT_SUPPORTED || MIN_IE_VERSION >= 10) && ENVIRONMENT == 'web'
var decodeBase64 = atob;
#else
// IE 10+ have atob, and so do practically all other browsers. Older node (at
// least 14) does not, though, nor does d8, and support in workers requires
// somewhat newer versions of browsers. So if we don't support IE and we are on
// the web, we don't need a polyfill here, but otherwise we do.
// See https://developer.mozilla.org/en-US/docs/Web/API/atob
var decodeBase64 = typeof atob === 'function' ? atob : function (input) {
  .. polyfill code
```
For readers of the code, this waits until the else to actually explain things. But the emitted JS is clear.

Which would you prefer?

Reply:
having the comment located alongside the polyfill makes sense to me.

Reply:
Ok, done.